File: app/src/main/java/com/example/roadready/SignupDealership.java
package com.example.roadready;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;

import java.lang.reflect.Array;
import java.util.Arrays;
import java.util.List;

public class SignupDealership extends Login {
    private Spinner spinner;
    private Button lgnBtn2,sgnupBtn;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_signup_dealership);
        sgnupBtn = (Button) findViewById(R.id.sgnupBtn);

        sgnupBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openDealerHome();
            }
        });


        final List<String> gender = Arrays.asList("Gender", "Male", "Female");

        final Spinner spinner = findViewById(R.id.spinner);
        ArrayAdapter adapter = new ArrayAdapter(getApplicationContext(), android.R.layout.simple_spinner_item,gender);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);

        spinner.setAdapter(adapter);
        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
                String newItem = spinner.getSelectedItem().toString();
            }

            @Override
            public void onNothingSelected(AdapterView<?> adapterView) {

            }
        });
        lgnBtn2 = (Button) findViewById(R.id.lgnBtn2);
        lgnBtn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openSignup();
            }
        });


    }


}

File: app/src/main/java/com/example/roadready/SignupDealership.kt
package com.example.roadready

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.Spinner
import androidx.appcompat.app.AppCompatActivity
import java.util.*

class SignupDealership : AppCompatActivity() {
    private val spinner: Spinner? = null
    private var lgnBtn2: Button? = null
    private var sgnupBtn: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_signup_dealership)
        sgnupBtn = findViewById<View>(R.id.sgnupBtn) as Button
        sgnupBtn!!.setOnClickListener { openDealerHome() }
        val gender = Arrays.asList("Gender", "Male", "Female")
        val spinner = findViewById<Spinner>(R.id.spinner)
        val adapter: ArrayAdapter<*> = ArrayAdapter<Any?>(applicationContext, android.R.layout.simple_spinner_item, gender as List<Any?>)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner.adapter = adapter
        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(adapterView: AdapterView<*>?, view: View, i: Int, l: Long) {
                val newItem = spinner.selectedItem.toString()
            }

            override fun onNothingSelected(adapterView: AdapterView<*>?) {}
        }


        lgnBtn2 = findViewById<View>(R.id.lgnBtn2) as Button
        lgnBtn2!!.setOnClickListener { openLogin() }
    }


    fun openLogin() {
        val intent = Intent(this, Login::class.java)
        startActivity(intent)
    }
    @SuppressLint("SuspiciousIndentation")
    fun openDealerHome(){
        val intent = Intent(this, navigationactivity::class.java)
                startActivity(intent)
    }
}